src/common/src/collection/estimate_size.rs

```diff
@@ -61,3 +67,29 @@ impl EstimateSize for String {
         self.capacity()
     }
 }
+
+impl EstimateSize for () {
+    fn estimated_heap_size(&self) -> usize {
+        0
+    }
+}
+
+impl EstimateSize for &str {
+    fn estimated_heap_size(&self) -> usize {
+        self.len()
+    }
+}
```

Comment:
Seems there's already a trait
https://docs.rs/deepsize/latest/deepsize/trait.DeepSizeOf.html

Reply:
if we use this maybe can simplify a lot.

Reply:
This seems costly as it always sums up the children and is accurate. 🤔 BTW, we're thinking of deprecating this trait and using allocator API for this.

Reply:
Yes, but I found that most batch patterns will not insert/update the hashmap after built, so one costly access looks no bad. 

Allocator is good, but I'm not sure whether it can count all usages (e.g. DataChunk, HashMap, BTreeMap)? Or our previous task local allocator (if it can be live again 🥵)

Reply:
Another alternative for estimated size: https://docs.rs/memuse/latest/memuse/